src/material/slider/slider-thumb.ts

```diff
@@ -0,0 +1,281 @@
+ * @docs-private
+ */
+@Component({
+  selector: 'mat-slider-visual-thumb',
+  templateUrl: './slider-thumb.html',
+  styleUrls: ['slider-thumb.css'],
+  host: {
+    'class': 'mdc-slider__thumb mat-mdc-slider-visual-thumb',
+    '[class.mdc-slider__thumb--focused]': '_sliderInput?._isFocused',
+  },
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+  providers: [{provide: MAT_SLIDER_VISUAL_THUMB_TOKEN, useExisting: MatSliderVisualThumb}],
+})
+export class MatSliderVisualThumb
+  implements MatSliderVisualThumbInterface, AfterViewInit, OnDestroy
+{
+  /** Whether the slider displays a numeric value label upon pressing the thumb. */
+  @Input() discrete: boolean;
+
+  /** Indicates which slider thumb this input corresponds to. */
+  @Input() thumbPosition: Thumb;
+
+  /** The display value of the slider thumb. */
+  @Input() valueIndicatorText: string;
+
+  /** The MatRipple for this slider thumb. */
+  @ViewChild(MatRipple) readonly _ripple: MatRipple;
+
+  /** The slider thumb knob. */
+  @ViewChild('knob') _knob: ElementRef<HTMLElement>;
+
+  /** The slider thumb value indicator container. */
+  @ViewChild('valueIndicatorContainer')
+  _valueIndicatorContainer: ElementRef<HTMLElement>;
+
+  /** The slider input corresponding to this slider thumb. */
+  private _sliderInput: MatSliderThumbInterface;
+
+  /** The native html element of the slider input corresponding to this thumb. */
+  private _sliderInputEl: HTMLInputElement;
+
+  /** The RippleRef for the slider thumbs hover state. */
+  private _hoverRippleRef: RippleRef | undefined;
+
+  /** The RippleRef for the slider thumbs focus state. */
+  private _focusRippleRef: RippleRef | undefined;
+
+  /** The RippleRef for the slider thumbs active state. */
+  private _activeRippleRef: RippleRef | undefined;
+
+  /** Whether the slider thumb is currently being hovered. */
+  private _isHovered: boolean = false;
+
+  /** Whether the slider thumb is currently being pressed. */
+  _isActive = false;
+
+  /** Whether the value indicator tooltip is visible. */
+  _isValueIndicatorVisible: boolean = false;
+
+  /** The host native HTML input element. */
+  _hostElement: HTMLElement;
+
+  constructor(
+    readonly _cdr: ChangeDetectorRef,
+    private readonly _ngZone: NgZone,
+    _elementRef: ElementRef<HTMLElement>,
+    @Inject(MAT_SLIDER_TOKEN) private _slider: MatSliderInterface,
+  ) {
+    this._hostElement = _elementRef.nativeElement;
+  }
+
+  ngAfterViewInit() {
+    this._ripple.radius = 24;
+    this._sliderInput = this._slider._getInput(this.thumbPosition)!;
+    this._sliderInputEl = this._sliderInput._hostElement;
+
+    // These listeners don't update any data bindings so we bind them outside
+    // of the NgZone to prevent Angular from needlessly running change detection.
+    this._ngZone.runOutsideAngular(() => {
+      this._sliderInputEl.addEventListener('pointermove', this._onPointerMove);
```

Comment:
Consider pulling `this._sliderInputEl` out into a variable since it's repeated in several places. Same goes for `ngOnDestroy` below.